canonical/mlmd-operator
# refactor: rewrite mlmd charm to sidecar pattern

This commit introduces changes that support the refactoring of the charm from podspec to sidecar pattern.

Fixes #71

#### Manual testing

1. Build and deploy the charm from this branch
2. Exec into the container and ensure the `/data` directory is there and has contents

```
$ kubectl exec -nmlmd-test mlmd-0 -c mlmd-grpc-server -ti  -- /bin/bash
root@mlmd-0:/# ls /data/
mlmd.db
```
3. Ensure the `metadata-grpc-service` exists and it exposes port 8080

```
$ kubectl get svc -A | grep metadata-grpc-service
mlmd-test         metadata-grpc-service   ClusterIP   10.152.183.118   <none>        8080/TCP                 10m
```
4. Ensure you can change the `port` config

```
$ kubectl get svc -A | grep metadata-grpc-service
mlmd-test         metadata-grpc-service   ClusterIP   10.152.183.118   <none>        8080/TCP                 10m
$ juju config mlmd port="9090"
$ kubectl get svc -A | grep metadata-grpc-service
mlmd-test         metadata-grpc-service   ClusterIP   10.152.183.118   <none>        9090/TCP                 11m
```

5. Execute an upgrade from mlmd 1.14/stable to the version in this branch
